src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -710,7 +717,16 @@ private void Serialize(string filename)
                     var folder = Path.GetDirectoryName(filename);
                     if (!Directory.Exists(folder))
                     {
-                        Directory.CreateDirectory(folder);
+                        try
+                        {
+                            Directory.CreateDirectory(folder);
+                        }
+                        catch (UnauthorizedAccessException)
+                        {
+                            // service accounts won't be able to create directory
```

Comment:
This usage is an edge case, I'm not sure how much we want to optimize particularly as one can use the XDG_*_HOME env vars.  Also, I don't think it's guaranteed the user account has write access to $env:TEMP so we'd have to check that anyways.